Description: Convert all sh scripts to bash.
Author: Mateusz Łukasik <mati75@linuxmint.pl>
Bug-Debian: https://bugs.debian.org/772365

--- a/scripts/simpleburn-detect.sh
+++ b/scripts/simpleburn-detect.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 function detect() {
 	device=$1 #assumes it is a valid CD / DVD device
--- a/scripts/simpleburn-gauges.sh
+++ b/scripts/simpleburn-gauges.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 function mencodergauge () {
 	totalsize=$1 #bytes
--- a/scripts/simpleburn.sh
+++ b/scripts/simpleburn.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 function printusage() {
 	echo "error: invalid parameters"
